File: web/HelloWorld/src/HelloWorldLogic.ts
namespace HelloWorld {

    export class HelloWorldLogic {

        public displayMessage: fm.liveswitch.IAction1<string>;
        public saveFile: fm.liveswitch.IAction2<string, Uint8Array>;

        // <SingletonPattern>
        private static instance: HelloWorldLogic;

        public static getInstance(): HelloWorldLogic {
            if (HelloWorldLogic.instance == null) {
                HelloWorldLogic.instance = new HelloWorldLogic();
            }
            return HelloWorldLogic.instance;
        }
        // </SingletonPattern>

        private applicationId: string = Config.applicationId;
        private channelId: string = Config.channelId;
        private gatewayUrl: string = Config.gatewayUrl;
        private sharedSecret: string = Config.sharedSecret;
    }

}

File: ios/HelloWorld/HelloWorld/HelloWorldLogic.swift

import Foundation

class HelloWorldLogic : NSObject {
    static let instance = HelloWorldLogic()

    let _applicationId: String = Config.applicationId
        let _channelId: String = Config.channelId
        let _gatewayUrl: String = Config.gatewayUrl
        let _sharedSecret: String = Config.sharedSecret
        
        func generateToken(claims: NSMutableArray) -> String {
            return FMLiveSwitchToken.generateClientRegister(
                withApplicationId: _applicationId,
                userId: (_client?.userId())!,
                deviceId: (_client?.deviceId())!,
                clientId: _client?.id(),
                clientRoles: nil,
                channelClaims: claims,
                sharedSecret: _sharedSecret)
        }
    
}
